feature/experimental/tunnel/otg_tests/tunnel_interface_based_disable_enable_test/tunnel_interface_disable_enable_test.go

```diff
@@ -0,0 +1,1558 @@
+				Value: &gpb.TypedValue_AsciiVal{
+					AsciiVal: config,
+				},
+			},
+		}},
+	}
+	return gpbSetRequest, nil
+}
+
+// Configure network instance
+func configureNetworkInstance(t *testing.T, dut *ondatra.DUTDevice) {
+
+	dutConfPath := gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut))
+	gnmi.Replace(t, dut, dutConfPath.Type().Config(), oc.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+}
+
+func ConfigureLoobackInterfaceWithIPv6address(t *testing.T, address string, dut *ondatra.DUTDevice) {
+
+	// IPv6 address on lo0 interface
+	t.Logf("Push the IPv4 address to lo0 interface :\n%s", dut.Vendor())
+	var config string
+	switch dut.Vendor() {
+	case ondatra.JUNIPER:
+		config = ConfigureAdditionalIPv6AddressonLoopback(address)
+		t.Logf("Push the CLI config:\n%s", config)
+	default:
+		t.Errorf("Invalid IPv6 Loop back address configuration")
+	}
+	gnmiClient := dut.RawAPIs().GNMI().Default(t)
+	gpbSetRequest, err := buildCliConfigRequest(config)
+	if err != nil {
+		t.Fatalf("Cannot build a gNMI SetRequest: %v", err)
+	}
+	t.Log("gnmiClient Set CLI config")
+	if _, err = gnmiClient.Set(context.Background(), gpbSetRequest); err != nil {
+		t.Fatalf("gnmiClient.Set() with unexpected error: %v", err)
+	}
+}
+
+func ConfigureLoobackInterfaceWithIPv4address(t *testing.T, address string, dut *ondatra.DUTDevice) {
+
+	// IPv4 address on lo0 interface
+	t.Logf("Push the IPv4 address to lo0 interface :\n%s", dut.Vendor())
+	var config string
+	switch dut.Vendor() {
+	case ondatra.JUNIPER:
+		config := ConfigureAdditionalIPv4AddressonLoopback(address)
+		t.Logf("Push the CLI config:\n%s", config)
+	default:
+		t.Errorf("Invalid IPv4 Loop back address configuration")
+	}
+	gnmiClient := dut.RawAPIs().GNMI().Default(t)
+	gpbSetRequest, err := buildCliConfigRequest(config)
+	if err != nil {
+		t.Fatalf("Cannot build a gNMI SetRequest: %v", err)
+	}
+	t.Log("gnmiClient Set CLI config")
+	if _, err = gnmiClient.Set(context.Background(), gpbSetRequest); err != nil {
+		t.Fatalf("gnmiClient.Set() with unexpected error: %v", err)
+	}
+
+}
+func ConfigureTunnelTermination(t *testing.T, intf *ondatra.Port, dut *ondatra.DUTDevice) {
+
+	// IPv4/IPv6 tunnel termination on underlay port
+	t.Logf("IPv4/IPv6 tunnel termination on underlay port :\n%s", dut.Vendor())
+	var config string
+	switch dut.Vendor() {
+	case ondatra.JUNIPER:
+		config = ConfigureTunnelTerminationOption(intf.Name())
+		t.Logf("Push the CLI config:\n%s", config)
+	default:
+		t.Errorf("Invalid Tunnel termination configuration")
+	}
+	gnmiClient := dut.RawAPIs().GNMI().Default(t)
+	gpbSetRequest, err := buildCliConfigRequest(config)
+	if err != nil {
+		t.Fatalf("Cannot build a gNMI SetRequest: %v", err)
+	}
+
+	t.Log("gnmiClient Set CLI config")
+	if _, err = gnmiClient.Set(context.Background(), gpbSetRequest); err != nil {
+		t.Fatalf("gnmiClient.Set() with unexpected error: %v", err)
+	}
+}
+
+func TestFtiTunnels(t *testing.T) {
+
+	p := &parameters{
```

Comment:
This massive parameters type looks unnecessary. Just define them as constants local to this test function, and feel free to inline the ones that are used only once

Reply:
Hmm, I see how other functions accept *parameters. Let me think about this one.